src/test/java/com/xwiki/authentication/saml/samlauth/SamlAuthenticatorTest.java

```diff
@@ -0,0 +1,678 @@
+        .then()
+            .shouldRedirectToIdentityProviderAndThenReturnTo("https://happy");
+    }
+
+    @Test
+    public void whenLoggedUserInByCookieAccess_ShouldReturnTheSameLoggedInUser() throws XWikiException, ComponentLookupException {
+        given()
+            .userIsLoggedInByCookie("ArthurDent")
+        .whenAuthenticationIsVerified()
+        .then()
+            .authenticatedUserIs("ArthurDent");
+    }
+
+    @Test
+    public void whenUserIsAlreadyLoggedInTheSession_ShouldReturnSameLoggedUser() throws Exception {
+        given()
+            .userIsLoggedInTheSession("ArthurDent")
+        .whenAuthenticationIsVerified()
+        .then()
+            .authenticatedUser("ArthurDent");
+    }
+
+    @Test
+    public void whenAnonymousUserAccessWithLoginAction_ShouldBeHandledByDefaultHandler() throws XWikiException, ComponentLookupException {
+        given()
+             .userIsAnonymous()
+             .accessWithGivenAction("login")
+        .whenUserAccess()
+        .then()
+             .verifyDefaultHandlerIsInvoked();
+    }
+
+    @Test
+    public void whenAnonymousUserAccessWithSkinAction_ShouldBeHandledByDefaultHandler() throws XWikiException, ComponentLookupException {
+        given()
+                .userIsAnonymous()
+                .accessWithGivenAction("skin")
+        .whenUserAccess()
+        .then()
+                .verifyDefaultHandlerIsInvoked();
+    }
+
+    @Test
+    public void whenAnonymousUserAccessWithSsxAction_ShouldBeHandledByDefaultHandler() throws XWikiException, ComponentLookupException {
+        given()
+            .userIsAnonymous()
+            .accessWithGivenAction("ssx")
+        .whenUserAccess()
+        .then()
+            .verifyDefaultHandlerIsInvoked();
+    }
+
+    @Test
+    public void whenAnonymousUserAccessWithLogoutAction_ShouldBeHandledByDefaultHandler() throws XWikiException, ComponentLookupException {
+        given()
+            .userIsAnonymous()
+            .accessWithGivenAction("logout")
+        .whenUserAccess()
+        .then()
+            .verifyDefaultHandlerIsInvoked();
+    }
+
+    @Test
+    public void whenAnonymousUserAccessWithLoginSubmit_ShouldBeHandledByDefaultHandler() throws XWikiException, ComponentLookupException {
+        given()
+            .userIsAnonymous()
+            .accessWithGivenAction("loginsubmit")
+        .whenUserAccess()
+        .then()
+            .verifyDefaultHandlerIsInvoked();
+    }
+
+    @Test
+    public void whenSamlAuthPresentAndSamlAcceptsAuthenticationAndUserExists_ShouldReturnExistingUser() throws Exception {
+        given()
+            .identityProviderAuthenticatedUser(user -> {
+                user.id = "arthur.dent@dontpanic.com";
+                user.firstName = "Arthur";
+                user.lastName = "Dent";
+            })
+            .defaultGroupForNewUsers("UserGroup")
+            .xwiki(users ->
+                users.userExists("ArthurDent")
+            )
```

Comment:
The method signature (and the IDEs hinting) says this receives a "configuration", so we suggest renaming the variable to:
```suggestion
            .xwiki(config ->
                config.userExists("ArthurDent")
            )
```

Same for any similar cases.